jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Query.java

```diff
@@ -241,6 +256,41 @@ public String makeTypeNameValueString(List<String> typeNames, String typeNameStr
 		return ImmutableList.of();
 	}
 
+	public synchronized boolean isNotificationListenerRegistered(JMXConnection jmxConnection) {
```

Comment:
Synchronization is probably not necessary here. The Query itself is immutable, so you're not protecting any state that is local to that class.